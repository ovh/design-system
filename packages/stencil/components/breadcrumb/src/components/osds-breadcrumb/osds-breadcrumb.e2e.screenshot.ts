import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
import { OdsBreadcrumbAttributes } from '@ovhcloud/ods-core';

describe('e2e:osds-breadcrumb', () => {
  let page: E2EPage;
  let el: E2EElement;

  async function setup({ onPage }: { attributes?: Partial<OdsBreadcrumbAttributes>; html?: string; onPage?: ({ page }: { page: E2EPage }) => void } = {}) {
    page = await newE2EPage();
    onPage && onPage({ page });
    await page.setContent(`
    <osds-breadcrumb>
    <osds-breadcrumb-item>
      <osds-link color="primary" href="home"> Home </osds-link>
    </osds-breadcrumb-item>
    <osds-breadcrumb-item><osds-link color="primary" href="web">Web</osds-link></osds-breadcrumb-item>
    <osds-breadcrumb-item><osds-text>Domain</osds-text></osds-breadcrumb-item>
  </osds-breadcrumb>
  <osds-breadcrumb>
  <osds-breadcrumb-item>
    <osds-link color="primary" href="home"> Home </osds-link>
  </osds-breadcrumb-item>
  <osds-breadcrumb-item><osds-link color="primary" href="web">Web</osds-link></osds-breadcrumb-item>
  <osds-breadcrumb-item><osds-link color="primary" href="Domains">Domains</osds-link></osds-breadcrumb-item>
  <osds-breadcrumb-item><osds-link color="primary" href="Informations générales">General information</osds-link></osds-breadcrumb-item>
  <osds-breadcrumb-item><osds-text>DNS zone</osds-text></osds-breadcrumb-item>
</osds-breadcrumb>
    `);
    await page.evaluate(() => document.body.style.setProperty('margin', '0px'));
    el = await page.find('osds-breadcrumb');
  }
  const screenshotActions = [
    {
      actionDescription: 'no action',
      action: () => {
        // noop
      },
    },
  ];
  describe('screenshots', () => {
    screenshotActions.forEach(({ actionDescription, action }) => {
      it(actionDescription, async () => {
        await setup({});
        action();
        await page.waitForChanges();

        await page.evaluate(() => {
          const element = document.querySelector('osds-breadcrumb') as HTMLElement;
          return { width: element.clientWidth, height: element.clientHeight };
        });
        await page.setViewport({ width: 600, height: 600 });
        const results = await page.compareScreenshot({ fullPage: false, omitBackground: true });
        expect(results).toMatchScreenshot({ allowableMismatchedRatio: 0.2 });
      });
    });
  });
});
