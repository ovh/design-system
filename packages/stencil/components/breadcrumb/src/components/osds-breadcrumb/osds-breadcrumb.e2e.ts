import { newE2EPage, E2EPage, E2EElement } from '@stencil/core/testing';

describe('osds-breadcrumb', () => {
  let page: E2EPage;
  let breadcrumb: E2EElement;
  let ellipsis: E2EElement;

  let activeElementId: string | undefined;

  async function setup({ html = '' }: { html?: string } = {}) {
    page = await newE2EPage();
    await page.setContent(`
      <osds-breadcrumb>
        ${html}
      </osds-breadcrumb>
    `);
    await page.evaluate(() => document.body.style.setProperty('margin', '4px'));
    breadcrumb = await page.find('osds-breadcrumb');
    ellipsis = await page.find('osds-breadcrumb >>> osds-icon[name="ellipsis"]');
  }

  async function updateReferences() {
    activeElementId = await page.evaluate(() => document.activeElement?.id);
  }

  describe('if we have 5 items', () => {
    beforeEach(async () => {
      await setup({
        html: `
        <osds-breadcrumb-item>
        <osds-link color="primary" href="home">Home</osds-link>
      </osds-breadcrumb-item>
      <osds-breadcrumb-item><osds-link color="primary" href="item1">Item 1</osds-link></osds-breadcrumb-item>
      <osds-breadcrumb-item><osds-link color="primary" href="item2">Item 2</osds-link></osds-breadcrumb-item>
      <osds-breadcrumb-item><osds-link color="primary" href="item3">Item 3</osds-link></osds-breadcrumb-item>
      <osds-breadcrumb-item><osds-text>Item active</osds-text></osds-breadcrumb-item>
        `,
      });
    });
    it('checks if the items are correctly displayed when clicking on the ellipsis', async () => {
      await page.waitForChanges(); // Wait for component to render

      expect(breadcrumb).not.toBeNull();
      expect(ellipsis).not.toBeNull();
      expect(breadcrumb.getAttribute('collapsed')).toBeTruthy;
      ellipsis.click();
      await page.waitForChanges(); // Wait for component to render
      expect(breadcrumb.getAttribute('collapsed')).toBeFalsy;
    });
  });
  describe('if we have 5 items and test with keyboard', () => {
    beforeEach(async () => {
      await setup({
        html: `
        <osds-breadcrumb-item>
        <osds-link color="primary" href="home">Home</osds-link>
      </osds-breadcrumb-item>
      <osds-breadcrumb-item><osds-link color="primary" href="item1">Item 1</osds-link></osds-breadcrumb-item>
      <osds-breadcrumb-item><osds-link color="primary" href="item2">Item 2</osds-link></osds-breadcrumb-item>
      <osds-breadcrumb-item><osds-link color="primary" href="item3">Item 3</osds-link></osds-breadcrumb-item>
      <osds-breadcrumb-item><osds-text>Item active</osds-text></osds-breadcrumb-item>
        `,
      });
      await updateReferences();
    });

    it('item should be focusable', async () => {
      await page.waitForChanges(); // Wait for component to render
    });

    it('checks whether the items are displayed when the Enter key is pressed on the ellipsis', async () => {
      await page.waitForChanges(); // Wait for component to render

      expect(breadcrumb).not.toBeNull();
      expect(ellipsis).not.toBeNull();
      expect(breadcrumb.getAttribute('collapsed')).toBeTruthy;

      // Simulate Enter key press on ellipsis
      await ellipsis.press('Enter');

      await page.waitForChanges(); // Wait for component to render

      expect(breadcrumb.getAttribute('collapsed')).toBeFalsy;
    });
    it('checks whether the items are displayed when the click is on the ellipsis', async () => {
      await page.waitForChanges(); // Wait for component to render

      expect(breadcrumb).not.toBeNull();
      expect(ellipsis).not.toBeNull();
      expect(breadcrumb.getAttribute('collapsed')).toBeTruthy;

      // Simulate Enter key press on ellipsis
      await ellipsis.click();

      await page.waitForChanges(); // Wait for component to render

      expect(breadcrumb.getAttribute('collapsed')).toBeFalsy;
    });
  });
});
