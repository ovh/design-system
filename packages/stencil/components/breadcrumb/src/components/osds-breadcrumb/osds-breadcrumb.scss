@import 'styles/osds-breadcrumb.mixins';
@import 'styles/osds-breadcrumb.color';
@import 'styles/osds-breadcrumb.size';
@import 'styles/osds-breadcrumb.typography';
@import '~@ovhcloud/ods-theming/color/ods-theming-color';
@import '~@ovhcloud/ods-theming/ods-theme';

// CSS targeted only for this component
// /!\ for theming purposes, it has to be done in theming files
// (i.e. packages/libraries/theming/...)
:host {

}

:host([attribute]) {

}

// apply the theme template for the component
@include ods-theme-component() {
  @include osds-breadcrumb-theme-color();
  @include osds-breadcrumb-theme-size();
  @include osds-breadcrumb-theme-typography();
}
