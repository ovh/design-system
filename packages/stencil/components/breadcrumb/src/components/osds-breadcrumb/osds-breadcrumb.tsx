import { Component, Element, Host, h, Listen, Prop, State, Watch } from '@stencil/core';
import {
  OdsBreadcrumb,
  OdsBreadcrumbAttributeItem,
  OdsBreadcrumbController,
  odsBreadcrumbDefaultAttributes,
  OdsBreadcrumbEvents,
  OdsBreadcrumbItemAttributes,
  OdsBreadcrumbMethods,
  OdsLogger,
} from '@ovhcloud/ods-core';
import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libraries/stencil-core';

/**
 * @slot (unnamed) - Breadcrumb content
 */
@Component({
  tag: 'osds-breadcrumb',
  styleUrl: 'osds-breadcrumb.scss',
  shadow: true
})
export class OsdsBreadcrumb implements OdsBreadcrumb<OdsStencilMethods<OdsBreadcrumbMethods>, OdsStencilEvents<OdsBreadcrumbEvents>> {
  private logger = new OdsLogger('OsdsBreadcrumb');
  private breadcrumbItems: OdsBreadcrumbItemAttributes[] = [];
  private parsedItems: OdsBreadcrumbAttributeItem[] = [];
  controller: OdsBreadcrumbController = new OdsBreadcrumbController(this);
  @Element() el!: HTMLElement;

  @State() isCollapsed = true;

  /** @see OdsBreadcrumbAttributes.contrasted */
  @Prop({ reflect: true }) public contrasted?: boolean = odsBreadcrumbDefaultAttributes.contrasted;

  /** @see OdsBreadcrumbAttributes.items */
  @Prop({ reflect: true }) public items: OdsBreadcrumbAttributeItem[] | string = [];

  componentWillLoad() {
    this.parseItems();
    this.updateBreadcrumb();
  }

  @Watch('items')
  onItemsChange() {
    this.parseItems();
    this.updateBreadcrumb();
  }

  /**
   * @see OdsBreadcrumbBehavior.onBreadcrumbItemCollapsedClick
   */
  @Listen('odsBreadcrumbItemCollapsedClick')
  onBreadcrumbItemCollapsedClick() {
    this.isCollapsed = false;
    this.updateBreadcrumb();
  }

  private parseItems() {
    if (typeof this.items === 'string') {
      try {
        this.parsedItems = JSON.parse(this.items);
      } catch {
        this.logger.warn('[OsdsBreadcrumb] items string could not be parsed.');
        this.parsedItems = [];
      }
    } else {
      this.parsedItems = [...this.items];
    }
  }

  private updateBreadcrumb() {
    this.breadcrumbItems = this.controller.getBreadcrumbItems(this.parsedItems, this.isCollapsed);
  }

  render() {
    return (
      <Host role="navigation">
        {
          (this.breadcrumbItems || []).map((breadcrumbItem, index) => (
            <osds-breadcrumb-item key={index}
                                  contrasted={this.contrasted}
                                  {...breadcrumbItem}>
            </osds-breadcrumb-item>
          ))
        }
      </Host>
    );
  }
}
