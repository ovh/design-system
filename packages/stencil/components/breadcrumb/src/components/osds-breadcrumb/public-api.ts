export { OsdsBreadcrumb } from './osds-breadcrumb'
