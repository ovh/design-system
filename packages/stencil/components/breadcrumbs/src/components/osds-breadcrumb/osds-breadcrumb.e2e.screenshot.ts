import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
import {
  OdsComponentAttributes2StringAttributes,
  OdsBreadcrumbsAttributes,
  odsBreadcrumbsDefaultAttributes,
} from '@ovhcloud/ods-core';
import { OdsCreateAttributes, OdsStringAttributes2Str, odsBreadcrumbsBaseAttributes } from '@ovhcloud/ods-testing';

import { OdsThemeColorIntentList } from '@ovhcloud/ods-theming';

describe('e2e:osds-breadcrumb', () => {
  let page: E2EPage;
  let el: E2EElement;

  async function setup({ attributes = {}, html = `` }: { attributes?: Partial<OdsBreadcrumbsAttributes>, html?: string }) {
    const minimalAttributes: OdsBreadcrumbsAttributes = OdsCreateAttributes(attributes, odsBreadcrumbsBaseAttributes);
    const stringAttributes = OdsComponentAttributes2StringAttributes<OdsBreadcrumbsAttributes>(minimalAttributes, odsBreadcrumbsDefaultAttributes);

    page = await newE2EPage();
    await page.setContent(`<osds-breadcrumb ${OdsStringAttributes2Str(stringAttributes)}>${html}</osds-breadcrumb>`);
    await page.evaluate(() => document.body.style.setProperty('margin', '0px'));

    el = await page.find('osds-breadcrumb');
  }

  describe('screenshots', () => {
    // Todo : add active behaviour on top of hover and focus
    [() => {}, () => el.setProperty('contrasted', true), () => el.setProperty('disabled', true), () => {
      el.setProperty('contrasted', true);
      el.setProperty('disabled', true);
    }].forEach((action) => {
      [() => {}, () => el.hover(), () => el.focus()].forEach((behaviour) => {
        OdsThemeColorIntentList.forEach((color) => {
          it([color, action, behaviour].join(', '), async () => {
            await setup({
              attributes: {
                color,
              },
              html: `Button`,
            });
            action();
            await behaviour();
            await page.waitForChanges();

            await page.evaluate(() => {
              const element = document.querySelector('osds-breadcrumb') as HTMLElement;
              return { width: element.clientWidth, height: element.clientHeight };
            });
            await page.setViewport({ width: 600, height:600 });
            const results = await page.compareScreenshot('button', { fullPage: false, omitBackground: true });
            expect(results).toMatchScreenshot({ allowableMismatchedRatio: 0 });
          });

          it(`${[color, action, behaviour].join(', ')} breadcrumbs`, async () => {
            await setup({
              attributes: {
                color,
                href: '#',
              },
              html: `Button`,
            });
            action();
            await behaviour();
            await page.waitForChanges();
            const results = await page.compareScreenshot('button', { fullPage: false, omitBackground: true });
            expect(results).toMatchScreenshot({ allowableMismatchedRatio: 0 });
          });
        });
      });
    });
  });
});
