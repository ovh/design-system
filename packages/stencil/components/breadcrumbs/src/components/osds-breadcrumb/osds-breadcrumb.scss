@import 'styles/osds-breadcrumb.mixins';
@import 'styles/osds-breadcrumb.color';
@import 'styles/osds-breadcrumb.typography';
@import '~@ovhcloud/ods-theming/color/ods-theming-color';
@import '~@ovhcloud/ods-theming/ods-theme';

// CSS for the component but not for theming

:host {


  }

  osds-breadcrumb-item {
    background-color: red;

  }


