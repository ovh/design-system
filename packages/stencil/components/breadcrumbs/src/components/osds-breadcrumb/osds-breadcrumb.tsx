import { Component, Element, Prop, State, h, Host, Listen, Event, EventEmitter, Watch } from '@stencil/core';
import { OdsBreadcrumbs, OdsBreadcrumbsController, odsBreadcrumbsDefaultAttributes, OdsBreadcrumbsEvents, OdsBreadcrumbsMethods } from '@ovhcloud/ods-core';
import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libraries/stencil-core';

import { HTMLStencilElement } from '@stencil/core/internal';
import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';

/**
 * @slot start - Fixed start breadcrumbs content
 * @slot (unnamed) - Breadcrumbs content
 * @slot end - Fixed end breadcrumbs content
 */
@Component({
  tag: 'osds-breadcrumb',
  styleUrl: 'osds-breadcrumb.scss',
  shadow: true,
})
export class OsdsBreadcrumbs implements OdsBreadcrumbs<OdsStencilMethods<OdsBreadcrumbsMethods>, OdsStencilEvents<OdsBreadcrumbsEvents>> {
  controller: OdsBreadcrumbsController = new OdsBreadcrumbsController(this);

  @Element() host!: HTMLStencilElement;

  /** @see OdsBreadcrumbsAttributes.color */
  @Prop({ reflect: true }) public color?: OdsThemeColorIntent = odsBreadcrumbsDefaultAttributes.color;

  /** @see OdsBreadcrumbsAttributes.contrasted */
  @Prop({ reflect: true }) public contrasted?: boolean = odsBreadcrumbsDefaultAttributes.contrasted;

  /** @see OdsBreadcrumbsAttributes.disabled */
  @Prop({ reflect: true }) disabled?: boolean = odsBreadcrumbsDefaultAttributes.disabled;

  /** @see OdsBreadcrumbsAttributes.active */
  @Prop({ reflect: true }) active?: boolean = odsBreadcrumbsDefaultAttributes.active;

  /** @see OdsBreadcrumbsAttributes.href */
  @Prop({ reflect: true }) public href: string = odsBreadcrumbsDefaultAttributes.href;

  @State() activeChanged!: boolean;

  @State() collapsed!: boolean;

  @Prop() separator?: boolean | undefined;

  @Prop() maxItems?: number;

  @Prop() itemsBeforeCollapse = 1;

  @Prop() itemsAfterCollapse = 1;

  /**
   * @see OdsChipBehavior.beforeRender
   */
  beforeRender(): void {
    this.controller.validateAttributes();
  }

  componentWillRender(): void {
    this.beforeRender();
  }

  @Listen('collapsedClick')
  @Watch('maxItems')
  @Watch('itemsBeforeCollapse')
  @Watch('itemsAfterCollapse')
  render() {
    return (
      <Host role="navigation">
        <ul>
          <slot></slot>
        </ul>
      </Host>
    );
  }
}
