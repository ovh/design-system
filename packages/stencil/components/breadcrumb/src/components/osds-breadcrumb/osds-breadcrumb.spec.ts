import { OdsBreadcrumbAttributes, OdsBreadcrumbController, OdsComponentAttributes2StringAttributes, odsBreadcrumbDefaultAttributes } from '@ovhcloud/ods-core';
import { OdsCreateAttributes, OdsStringAttributes2Str, odsBreadcrumbBaseAttributes, odsUnitTestAttribute } from '@ovhcloud/ods-testing';
import { SpecPage, newSpecPage } from '@stencil/core/testing';

import { OdsThemeColorIntentList } from '@ovhcloud/ods-theming';
import { OsdsBreadcrumb } from './osds-breadcrumb';
import { getAttributeContextOptions } from '@ovhcloud/ods-stencil/libraries/stencil-testing';

describe('spec:osds-breadcrumb', () => {
  let page: SpecPage;
  let root: HTMLElement | undefined;
  let instance: OsdsBreadcrumb;
  let controller: OdsBreadcrumbController;
  let startSlot: HTMLElement;
  let ulElement: HTMLElement;

  afterEach(() => {
    jest.clearAllMocks();
  });

  async function setup({ attributes = {} }: { attributes?: Partial<OdsBreadcrumbAttributes> } = {}) {
    const minimalAttributes: OdsBreadcrumbAttributes = OdsCreateAttributes(attributes, odsBreadcrumbBaseAttributes);
    const stringAttributes = OdsComponentAttributes2StringAttributes<OdsBreadcrumbAttributes>(minimalAttributes, odsBreadcrumbDefaultAttributes);

    page = await newSpecPage({
      components: [OsdsBreadcrumb],
      html: `<osds-breadcrumb ${OdsStringAttributes2Str(stringAttributes)}>My Breadcrumb</osds-breadcrumb>`,
    });

    root = page.root;
    instance = page.rootInstance;
    startSlot = page.root?.shadowRoot?.querySelector('slot');
    ulElement = page.root?.shadowRoot?.querySelector('ul');
    await page.waitForChanges();
  }

  it('should render', async () => {
    await setup({});
    await page.waitForChanges();

    expect(root?.shadowRoot).toBeTruthy();
    expect(instance).toBeTruthy();
  });

  describe('content parent', () => {
    const config = {
      page: () => page,
      instance: () => instance,
      setup,
    };

    it('should have a slot', async () => {
      await setup({ attributes: {}, html: `<slot></slot>` });
      expect(startSlot).toBeTruthy();
    });

    it('should have a breadcrumb-item', async () => {
      await setup();
      expect(ulElement).toBeTruthy();
      expect(await page.root?.shadowRoot.querySelectorAll('osds-breadcrumb-item')).toBeTruthy();
    });

    xit('should toggle collapsed on link click', async () => {
      await setup();
      const link = await page.root?.shadowRoot.querySelector('osds-link');
      const toggleCollapsedSpy = jest.spyOn(instance, 'toggleCollapsed');

      link?.click();
      await page.waitForChanges();

      expect(toggleCollapsedSpy).toHaveBeenCalled();
    });

    xit('should emit odsBreadcrumbItemSelection event on link click', async () => {
      await setup();
      const link = await page.root?.shadowRoot.querySelector('osds-link');
      const odsBreadcrumbItemSelectionSpy = jest.spyOn(instance.odsBreadcrumbItemSelection, 'emit');

      link?.click();
      await page.waitForChanges();

      expect(odsBreadcrumbItemSelectionSpy).toHaveBeenCalled();
    });

    xit('should toggle collapsed on Enter or Space key press', async () => {
      await setup();
      const keyUpEvent = new KeyboardEvent('keyup', { key: 'Enter' });
      const toggleCollapsedSpy = jest.spyOn(instance, 'toggleCollapsed');

      page.doc.dispatchEvent(keyUpEvent);
      await page.waitForChanges();

      expect(toggleCollapsedSpy).toHaveBeenCalled();
    });

    xit('should emit odsBreadcrumbItemSelection event on Enter or Space key press', async () => {
      await setup();
      const keyUpEvent = new KeyboardEvent('keyup', { key: 'Enter' });
      const odsBreadcrumbItemSelectionSpy = jest.spyOn(instance.odsBreadcrumbItemSelection, 'emit');

      page.doc.dispatchEvent(keyUpEvent);
      await page.waitForChanges();

      expect(odsBreadcrumbItemSelectionSpy).toHaveBeenCalled();
    });

    xit('should emit odsBreadcrumbItemSelection event on Tab key press', async () => {
      await setup();
      const keyUpEvent = new KeyboardEvent('keyup', { key: 'Tab' });
      const odsBreadcrumbItemSelectionSpy = jest.spyOn(instance.odsBreadcrumbItemSelection, 'emit');

      page.doc.dispatchEvent(keyUpEvent);
      await page.waitForChanges();

      expect(odsBreadcrumbItemSelectionSpy).toHaveBeenCalled();
    });
  });

  describe('attributes', () => {
    const config = {
      page: () => page,
      instance: () => instance,
      setup,
    };

    describe('collapsed', () => {
      odsUnitTestAttribute<OdsBreadcrumbAttributes, 'collapsed'>({
        ...getAttributeContextOptions<OdsBreadcrumbAttributes, OsdsBreadcrumb, 'collapsed'>({
          name: 'collapsed',
          list: [false, true],
          defaultValue: odsBreadcrumbDefaultAttributes.collapsed,
          ...config,
        }),
      });
    });
  });
});
