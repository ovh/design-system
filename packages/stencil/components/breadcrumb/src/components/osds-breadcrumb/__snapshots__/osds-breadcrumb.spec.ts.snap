// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OsdsBreadcrumb should render 1`] = `
<osds-breadcrumb role="navigation">
  <mock:shadow-root>
    <ul>
      <slot></slot>
    </ul>
  </mock:shadow-root>
  <osds-breadcrumb-item>
    Home
  </osds-breadcrumb-item>
  <osds-breadcrumb-item>
    Category 1
  </osds-breadcrumb-item>
  <osds-breadcrumb-item>
    Category 2
  </osds-breadcrumb-item>
  <osds-breadcrumb-item>
    Product
  </osds-breadcrumb-item>
</osds-breadcrumb>
`;
